compiler/passes/src/function_inlining/function_inliner.rs

```diff
@@ -29,7 +27,7 @@ pub struct FunctionInliner<'a> {
     /// A wrapper around an Assigner used to create unique variable assignments.
     pub(crate) assignment_renamer: AssignmentRenamer<'a>,
     /// A map of reconstructed functions in the current program scope.
-    pub(crate) reconstructed_functions: IndexMap<Symbol, Function>,
+    pub(crate) reconstructed_functions: Vec<(Symbol, Function)>,
```

Comment:
`reconstructed_functions` in the function inliner should remain in an `IndexMap`